### PR TITLE
WIP - Feature/ci install from sources on centos7 #patch

### PR DESCRIPTION
**Pull request by MarkGibbons**
### Description

Add and update #227 
Move the needed package install from a test recipe to the cookbook source install recipe

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: test/cookbooks/test/recipes/source_centos7.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+node.default['yum']['centos-sclo']['enabled'] = true
+node.default['yum']['centos-sclo']['managed'] = true
+node.default['yum']['centos-sclo-rh']['enabled'] = true
+node.default['yum']['centos-sclo-rh']['managed'] = true
+include_recipe 'yum-centos'
+
+yum_package 'devtoolset-7-gcc-c++'
```
Comment (ramereth) [inline test/cookbooks/test/recipes/source_centos7.rb:7]: This really should be included in the source recipe for the cookbook so that it works "out of the box" for users. We can use an attribute to turn this on/off for air gap environments.

Comment (MarkGibbons) [reply test/cookbooks/test/recipes/source_centos7.rb]: I agree that the package should be installed in mainline code.  But, I don't think this cookbook is the right place to set up a a yum repository.   We could set up repository access for centos, but that's a case where we have network access to the centos repository.  Building from source is probably a thing to do when network access is not available so setting up the repository seems pointless. 

We also are not setting up access to repositories for other platforms. RHEL setup is outside of this cookbook.   For Debian we are are running apt-get commands outside the cookbook to make it work.  Centos7 is, for me, an edge case that doesn't need special handling in the main recipe.

Comment (ramereth) [reply test/cookbooks/test/recipes/source_centos7.rb]: We setup repositories in other cookbooks if we need them to get it working. I don't see why that can't be the case here, especially if we make it optional for air gapped environments. I think for most use cases, users would expect this to work out of the box without having to add more code to get it to work on EL7.

**Review by Xorima — APPROVED** [review]

**Comment by damacus** [comment]
@MarkGibbons if this is still an issue. Feel free to reopen 👍🏼
